core/engine.go

```diff
@@ -226,10 +220,8 @@ func (e *Engine) Run(ctx context.Context) error {
 	for {
 		select {
 		case <-ticker.C:
-			if len(sampleContainers) > 0 {
-				e.processSamples(sampleContainers)
-				sampleContainers = []stats.SampleContainer{}
-			}
+			e.processSamples(sampleContainers)
+			sampleContainers = []stats.SampleContainer{}
```

Comment:
I agree with the removals of `if` clauses above, when they're immediately followed by a `for ... range` loop, but in this case I think the `if` should stay, given that it would save us a function call and an allocation

Reply:
Agree.